src/nvim/os/shell.c

```diff
@@ -59,7 +104,8 @@ char **shell_build_argv(const char *cmd, const char *extra_args)
 
   if (cmd) {
     i += tokenize(p_shcf, rv + i);   // Split 'shellcmdflag'
-    rv[i++] = xstrdup(cmd);          // Push a copy of the command.
+    rv[i++] = shell_escape(cmd);     // Process command string with
```

Comment:
Doing this here will affect `tv_to_argv` which is used by `f_jobstart`--but only if the first arg is String, so that seems fine. It also affects `f_termopen()`, not sure about that (and I don't care too much because we should deprecate that function and enhance `jobstart()` to replace it).

So, 👍